### ISSUE TITLE
jiralib-user-login-name is always nil

### ISSUE THREAD

**Issue by lvsv** [ahungry/org-jira#129]
I found that `jiralib-user-login-name` is always `nil` and `jiralib-worklog-import--filters-alist` doesn't work for me.

Something is wrong in the `jiralib-login` function:
```
 (if jiralib-use-restapi
        (setq jiralib-token `("Authorization" . , (format "Basic %s" (base64-encode-string (concat username ":" password)))))
    (unless jiralib-wsdl
      (jiralib-load-wsdl))
    (setq jiralib-token
          (car (soap-invoke jiralib-wsdl "jirasoapservice-v2" "login" username password)))
    (setq jiralib-user-login-name username)))
```
Don't know what is the true here. If `jiralib-use-restapi` is true, should we set `jiralib-token` and execute `jiralib-load-wsdl`, and if it is the false, we should set `jiralib-token` as `(car (soap-invoke...`, and without any conditions always set `jiralib-user-login-name` as `username`?

**Comment by ahungry** [comment]
jiralib-user-login-name was a remnant from the old soap / wsdl stuff which is basically deprecated but not cleaned out at this point.  I failed to notice that when this was added (the filter worklog feature) it was referencing it.  I think a better option would be for it to just be a user defcustom, as a lot of people hard code the jiralib-token to work around Oauth providers when their jira's basic auth is disabled (if we had features relying on it being set, but it only got set during the interactive login, that would be bad).
